src/test/unit_tests/test_notifications.js

```diff
@@ -0,0 +1,203 @@
+                const input = Buffer.concat(chunks);
+                const notif = JSON.parse(input.toString());
+
+                if (notif !== "test notification") {
+                    assert.strictEqual(notif.Records[0].s3.bucket.name, expected_bucket, 'wrong bucket name in notification');
+                    assert.strictEqual(notif.Records[0].eventName, expected_event_name, 'wrong event name in notification');
+                }
+
+                res.writeHead(200, {'Content-Type': 'text/plain'});
+                res.end();
+                server_done = true;
+            }).listen(9999);
+        });
+
+        mocha.after(() => {
+            http_server.close();
+        });
+
+        mocha.it('set/get notif conf s3ops', async () => {
+            await s3.putBucketNotificationConfiguration({
+                Bucket,
+                NotificationConfiguration: {
+                    TopicConfigurations: [{
+                        "Id": "system_test_http_no_event",
+                        "TopicArn": http_connect_filename,
+                    }],
+                },
+            });
+
+            const get = await s3.getBucketNotificationConfiguration({Bucket});
+            assert.strictEqual(get.TopicConfigurations[0].Id, 'system_test_http_no_event');
+        });
+
+        mocha.it('simple notif put', async () => {
+            await s3.putObject({
+                Bucket,
+                Key: 'f1',
+                Body: 'this is the body',
+            });
+
+            await notify_await_result({bucket: Bucket, event_name: 'ObjectCreated:Put'});
+        });
+
+
+        mocha.it('simple notif delete', async () => {
+            await s3.deleteObject({
+                Bucket,
+                Key: 'f1',
+            });
+
+            await notify_await_result({bucket: Bucket, event_name: 'ObjectRemoved:Delete'});
+        });
+
+
+        mocha.it('event notif', async () => {
+
+            const set = await s3.putBucketNotificationConfiguration({
+                Bucket,
+                NotificationConfiguration: {
+                    TopicConfigurations: [{
+                        "Id": "system_test_http_event",
+                        "TopicArn": http_connect_filename,
+                        "Events": ["s3:ObjectCreated:*"],
+                    }],
+                },
+            });
+
+            assert.strictEqual(set.$metadata.httpStatusCode, 200);
+
+            await s3.putObject({
+                Bucket,
+                Key: 'f2',
+                Body: 'this is the body',
+            });
+
+            await notify_await_result({bucket: Bucket, event_name: 'ObjectCreated:Put'});
+
+            await s3.deleteObject({
+                Bucket,
+                Key: 'f2',
+            });
+
+            //there shouldn't be a notification for the delete, wait 2 seconds to validate this
+            await notify_await_result({timeout: 2000});
+        });
+    });
+
+});
+
+const step_wait = 100;
```

Comment:
Why global variable?
It can be moved as a local variable in the function, right?

Reply:
Yes, but a const makes more sense. It's not a parameter.